spec/models/user_spec.rb

```diff
@@ -32,22 +33,29 @@
 
 require 'spec_helper'
 
-RSpec.describe User do
+describe User do
   it 'has a valid factory' do
     old_count = User.count
-    expect(FactoryGirl.create(:user)).to be_valid
-    FactoryGirl.create_list(:user, 9)
+    expect(create(:user)).to be_valid
+    create_list(:user, 9)
     expect(User.count).to eq old_count + 10
   end
 
-  it 'full_name returns correct name' do
-    # Email users
-    email_user = FactoryGirl.create(:user)
-    full_name = email_user.full_name
-    expect(full_name).to eq email_user.username
-    # Facebook users
-    fb_user = FactoryGirl.create(:user_with_fb)
-    full_name_fb = fb_user.full_name
-    expect(full_name_fb).to eq fb_user.first_name + ' ' + fb_user.last_name
+  context 'when user was created with Facebook' do
+    let!(:user) { create(:user) }
+    let(:full_name) { user.full_name }
+
+    it 'returns the correct name' do
+      expect(full_name).to eq user.username
+    end
+  end
+
+  context 'when user was created with regular login' do
+    let!(:user) { create(:user_with_fb) }
+    let(:full_name) { user.full_name }
+
+    it 'returns the correct name' do
+      expect(full_name).to eq user.first_name + ' ' + user.last_name
```

Comment:
Use string interpolation "#{user.first_name} #{user.last_name}`. It automatically runs the `to_s` method if it's necessary so it's considered safer than concatenate.

(I know that in this case we are dealing with strings)